File: 08_11_2025/Q2.java
/* 
Q2. Write a program in Java to count the number of inversions in a given array
Expected Output :
The given array is : 1 9 6 4 5
The inversions are: (9, 6) (9, 4) (9, 5) (6, 4) (6, 5)
The number of inversions can be formed from the array is: 5
*/

import java.util.*;

public class Q2 {
    public static void main(String  x[]) {
        Scanner sc = new Scanner(System.in);

        // Get array input
        System.out.println("Enter size of array:");
        int n = sc.nextInt();
        int a[] = new int[n];

        System.out.println("Enter " + n + " elements in array:");
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }

        // Display the given array
        System.out.print("The given array is : ");
        for (int i = 0; i < n; i++) {
            System.out.print(a[i] + " ");
        }
        System.out.println();

        int count = 0;

        System.out.print("The inversions are: ");
        // Check for inversions using nested loops
        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                if (a[i] > a[j]) {
                    System.out.print("(" + a[i] + ", " + a[j] + ") ");
                    count++;
                }
            }
        }

        System.out.println();
        System.out.println("The number of inversions can be formed from the array is: " + count);
    }
}

File: 20_02_2026/JS_Test/js/Q2.js
/*
Q2. Write a JavaScript program to count the frequency of each element in an integer array.
Input:
[2, 3, 2, 5, 3, 2]
Output:
2 → 3 times
3 → 2 times
5 → 1 time
Description:
Use nested loops to count occurrences. Mark visited elements manually
*/
let a = [2, 3, 2, 5, 3, 2];

//traverse all elements  for counting frequency
for (let i = 0; i < a.length; i++) {
  //check if element is not visited  then only count frequency
  if (a[i] !== -1) {
    let count = 1; //strt from 1

    //count freqeuncy and after count mark as visted by using -1
    for (let j = i + 1; j < a.length; j++) {
      if (a[i] === a[j]) {
        count++;
        a[j] = -1; // mark visited
      }
    }
    //print elemernt with its frequency
    console.log(a[i] + " => " + count + " times");
  }
}
